packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -1676,3 +1750,93 @@ abstract class MultiSelectableSelectionContainerDelegate extends SelectionContai
     return finalResult!;
   }
 }
+
+/// Calls [builder] with the [ContextMenuButtonItem]s representing the
+/// buttons in this platform's default text selection menu.
+///
+/// By default the [targetPlatform] will be [defaultTargetPlatform].
+///
+/// See also:
+///
+/// * [TextSelectionToolbarButtonsBuilder], which builds the button Widgets
+///   given [ContextMenuButtonItem]s.
+/// * [DefaultTextSelectionToolbar], which builds the toolbar itself.
+/// * [EditableTextContextMenuButtonItemBuilder], which performs a similar role
+///   but for [EditableText]'s context menu.
+class _SelectableRegionContextMenuButtonItemsBuilder extends StatelessWidget {
```

Comment:
@chunhtai, I'd prefer to make this public so that users can build the same buttons that show by default for SelectableRegion, but the delegate here is a private class _SelectableRegionState. I wonder what you think of this approach, if you think we could make _SelectableRegionState public or if there is another way.

This is the same pattern that I use to allow users to customize EditableText's default buttons. That class is EditableTextContextMenuButtonDatasBuilder, and it is passed EditableTextState, which is public.

Reply:
Yeah I think we can make it public if there is a reasonable use case.
I remember I originally made it public so you can see a lot of public method has doc string, but decide to close it until there is a use case for it.

Reply:
Ok good to hear, I'll plan to make that change to this PR.